src/components/molecules/Settings/SettingPage/index.tsx

```diff
@@ -18,7 +19,10 @@ const SettingPage: React.FC<Props> = ({
   const handleClick = () => {
     setIsOpen(o => !o);
   };
-
+  const handleKeyDown = useCallback(() => {
+    navigate(`/dashboard/${currentTeam?.id}`);
+  }, []);
```

Comment:
Personally I don't feel like this redirect is desirable. I would be confused if I hit Escape in the Settings and got redirected to the dashboard. Was this change discussed with @lavalse (Red)? 

Ignoring that for a moment, my understanding of this bug is that somewhere in the code (or a package we use) we are redirecting on the escape key. 
I took a quick look around the Settings but couldn't see anything obvious, so it looks like you'll need to investigate a bit to figure out where the problematic redirect happens, and stop it. Unless @lavalse has a different idea, I feel like NOT having a redirect on ESC is better than having one.